pkg/controllers/automigration/controller.go

```diff
@@ -377,43 +401,38 @@ func (c *Controller) getTotalAndReadyReplicas(
 	return totalReplicas, readyReplicas, nil
 }
 
-func (c *Controller) getPodsFromCluster(
-	ctx context.Context,
-	unsObj *unstructured.Unstructured,
-	clusterName string,
-) (*corev1.PodList, bool, error) {
-	labelSelector, err := utilunstructured.GetLabelSelectorFromPath(unsObj, c.typeConfig.Spec.PathDefinition.LabelSelector, nil)
+func (c *Controller) getDesiredReplicas(unsObj *unstructured.Unstructured) (int64, error) {
+	desiredReplicas, err := utilunstructured.GetInt64FromPath(unsObj, c.typeConfig.Spec.PathDefinition.ReplicasSpec, nil)
 	if err != nil {
-		return nil, false, fmt.Errorf("failed to get label selector from object: %w", err)
-	}
-	if labelSelector == nil {
-		return nil, false, fmt.Errorf("missing label selector on object")
+		return 0, fmt.Errorf("desired replicas: %w", err)
+	} else if desiredReplicas == nil {
+		return 0, fmt.Errorf("no desired replicas at %s", c.typeConfig.Spec.PathDefinition.ReplicasSpec)
 	}
 
-	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+	return *desiredReplicas, nil
+}
+
+func (c *Controller) getPodsFromCluster(
+	ctx context.Context,
+	unsClusterObj *unstructured.Unstructured,
+	clusterName string,
+) ([]*corev1.Pod, bool, error) {
+	plugin, err := plugins.ResolvePlugin(c.typeConfig)
 	if err != nil {
-		return nil, false, fmt.Errorf("failed to convert label selector to selector: %w", err)
+		return nil, false, fmt.Errorf("failed to get plugin for FTC: %w", err)
 	}
 
 	client, err := c.federatedInformer.GetClientForCluster(clusterName)
 	if err != nil {
-		return nil, false, fmt.Errorf("failed to get client for cluster: %w", err)
+		return nil, true, fmt.Errorf("failed to get client for cluster: %w", err)
```

Comment:
Why do we require backoff here?

Reply:
If we fail to get the client for cluster, it could mean that the federatedinformer is out of date. In this case, I think should retry until the federatedinformer becomes up to date?

Reply:
This is handled in other controllers by reconciling everything on `Cluster(A|Un)available`, which is wasteful and dangerous. Let's return true here and revisit this after we replace `FederatedInformer` with `FederatedInformerManager`